src/utils.ts

```diff
@@ -7,125 +16,157 @@ import defaultLeadLayout from "./resources/default_layout/lead.json";
+        : defaultLeadLayout;
+      return layout.view;
+    })
+    .with(["Account", "home"], () => {
+      const layout: AccountLayout = settings?.mapping_account
+        ? JSON.parse(settings.mapping_account)
+        : defaultAccountLayout;
+      return layout.home;
+    })
+    .with(["Account", "view"], () => {
+      const layout: AccountLayout = settings?.mapping_account
+        ? JSON.parse(settings.mapping_account)
+        : defaultAccountLayout;
+      return layout.view;
+    })
+    .with(["Task", "list"], () => {
+      const layout: TaskLayout = settings?.mapping_task
+        ? JSON.parse(settings.mapping_task)
+        : defaultTaskLayout;
 
-            return layout.home;
-        })
-        .with(["Contact", "view"], () => {
-            const layout: ContactLayout = settings?.mapping_contact
-                ? JSON.parse(settings.mapping_contact)
-                : defaultContactLayout
-            ;
-
-            return layout.view;
-        })
-        .with(["Note", "list"], () => {
-            const layout: NoteLayout = settings?.mapping_note
-                ? JSON.parse(settings.mapping_note)
-                : defaultNoteLayout
-            ;
+      return layout.list;
+    })
+    .with(["Task", "view"], () => {
+      const layout: TaskLayout = settings?.mapping_task
+        ? JSON.parse(settings.mapping_task)
+        : defaultTaskLayout;
 
-            return layout.list;
+      return layout.list;
+    })
+    .with(["Event", "list"], () => {
+      const layout: EventLayout = settings?.mapping_note
+        ? JSON.parse(settings.mapping_note)
+        : defaultEventLayout;
+      return layout.list;
+    })
+    .with(["Event", "view"], () => {
+      const layout: EventLayout = settings?.mapping_note
+        ? JSON.parse(settings.mapping_note)
+        : defaultEventLayout;
+      return layout.list;
+    })
+    .otherwise(() => null);
+}
+export const mapErrorMessage = (error: Error): string | null => {
+  const parsedErr = JSON.parse(error.message) as {
+    status: number;
+    message: string;
+  };
+  const parsedSalesforceErr = JSON.parse(parsedErr.message) as {
+    errorCode: string;
+    fields: string[];
+    message: string;
+  }[];
+  console.log(parsedSalesforceErr);
+  return parsedSalesforceErr.reduce((acc, curr) => {
+    return (
+      acc +
+      match([curr.message])
+        .with(["Event duration cannot be negative"], () => {
+          return "The dates are invalid, perhaps you have set the start date after the end date?\n\n";
         })
-        .with(["Note", "view"], () => {
-            const layout: NoteLayout = settings?.mapping_note
-                ? JSON.parse(settings.mapping_note)
-                : defaultNoteLayout
-                ;
-                
-                return layout.view;
+        .otherwise(() => {
+          return match([curr.errorCode, curr.fields])
+            .with(["REQUIRED_FIELD_MISSING", ["ActivityDateTime"]], () => {
```

Comment:
Hard coded field, `ActivityDateTime`? what happens if someone changes are removes this field in Salesforce? Is that even possible? If it's not possible to remove/change this field  in Salesforce then this is *probably* fine.